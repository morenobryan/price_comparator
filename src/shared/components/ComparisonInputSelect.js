// @flow
import React from 'react';
import { Field } from 'redux-form';
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';
import { textColor, selectedUnitButton, unitButton } from '../../shared/styles';
import type { KeyboardType, ReturnKeyType } from '../../types';

type Props = {
  input: {
    onChange: any => void,
    value: string,
  },
  meta: {
    touched: boolean,
    error: string,
    warning: string,
  },
  options: Array<string>,
  selectedOption: string,
  selectedColor: string,
  fieldTitle: string,
};

export default class ComparisonInputSelect extends React.Component<Props> {
  render() {
    const {
      options,
      selectedOption,
      selectedColor,
      input: { onChange, value, ...restInput },
      fieldTitle,
      ...pickerProps
    } = this.props;

    return (
      <View style={styles.container}>
        <Text style={styles.inputLabel}>{fieldTitle}</Text>
        <View style={styles.buttonsContainer}>
          {options.map(value => (
            <TouchableOpacity
              key={value}
              onPress={() => onChange(value)}
              style={
                selectedOption === value
                  ? [styles.unitButton, selectedUnitButton, { backgroundColor: selectedColor }]
                  : [styles.unitButton, unitButton]
              }
            >
              <Text style={styles.unitLabel}>{value}</Text>
            </TouchableOpacity>
          ))}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonsContainer: {
    flex: 0.5,
    flexDirection: 'row',
  },
  inputLabel: {
    flex: 0.5,
    color: textColor,
    textAlign: 'right',
    marginRight: 10,
    fontFamily: 'proximaNovaAltRegular',
  },
  input: {
    color: textColor,
    flex: 1,
    height: 35,
  },
  unitButton: {
    borderRadius: 15,
    borderWidth: 1,
    flex: 1,
    marginLeft: 5,
    marginRight: 5,
    marginTop: 5,
    paddingBottom: 4,
    paddingLeft: 7,
    paddingRight: 7,
    paddingTop: 2,
  },
  unitLabel: {
    fontFamily: 'proximaNovaAltRegular',
    alignSelf: 'center',
  },
  unitLabelDisabled: {
    color: '#F5F5F5',
  },
});
