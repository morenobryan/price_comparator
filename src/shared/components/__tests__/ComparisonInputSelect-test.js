import 'react-native';
import React from 'react';
import ComparisonInputSelect from '../ComparisonInputSelect';

import renderer from 'react-test-renderer';

const props = {
  input: {
    onChange: () => {},
    value: 'test',
  },
  meta: {
    touched: false,
    error: 'error',
    warning: undefined,
  },
  options: ['one', 'two'],
  selectedOption: 'one',
  selectedColor: '#fff',
  fieldTitle: 'title',
};

test('renders correctly', () => {
  const tree = renderer.create(<ComparisonInputSelect {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
