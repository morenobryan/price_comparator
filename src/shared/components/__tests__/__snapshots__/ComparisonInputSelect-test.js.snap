// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#3D3527",
        "flex": 0.5,
        "fontFamily": "proximaNovaAltRegular",
        "marginRight": 10,
        "textAlign": "right",
      }
    }
  >
    title
  </Text>
  <View
    style={
      Object {
        "flex": 0.5,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#3D3527",
          "borderRadius": 15,
          "borderWidth": 1,
          "elevation": 3,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
          "paddingBottom": 4,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "fontFamily": "proximaNovaAltRegular",
          }
        }
      >
        one
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFF2DE",
          "borderColor": "#3D3527",
          "borderRadius": 15,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
          "paddingBottom": 4,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "fontFamily": "proximaNovaAltRegular",
          }
        }
      >
        two
      </Text>
    </View>
  </View>
</View>
`;
